src/utils/parser/utils/closest.js

```diff
@@ -0,0 +1,126 @@
+// @flow
+
+import * as t from "babel-types";
+
+import { traverseAst } from "./ast";
+import { isLexicalScope, getMemberExpression } from "./helpers";
+
+import type { SourceText, Location } from "debugger-html";
+import type { NodePath, Node } from "babel-traverse";
+
+function getNodeValue(node: Node) {
+  if (t.isThisExpression(node)) {
+    return "this";
+  }
+
+  return node.name;
+}
+
+/**
+ * helps find member expressions on one line and function scopes that are
+ * often many lines
+ */
+type nodeContainsLocationParams = {
+  node: Node,
+  location: Location
+};
+
+function nodeContainsLocation({ node, location }: nodeContainsLocationParams) {
+  const { start, end } = node.loc;
+  const { line, column } = location;
+
+  if (start.line === end.line) {
```

Comment:
that's tough to say. the other cases are obviously really important. We probably need better docs